sql/planbuilder/priv.go

```diff
@@ -147,30 +147,39 @@ func convertPrivilegeLevel(privLevel ast.PrivilegeLevel) plan.PrivilegeLevel {
 	}
 }
 
+func (b *Builder) buildAuthenticatedUser(user ast.AccountWithAuth) plan.AuthenticatedUser {
+	authUser := plan.AuthenticatedUser{
+		UserName: convertAccountName(user.AccountName)[0],
+	}
+	if user.Auth1 != nil {
+		authUser.Identity = user.Auth1.Identity
+		if user.Auth1.Plugin == "mysql_native_password" && len(user.Auth1.Password) > 0 {
+			authUser.Auth1 = plan.AuthenticationMysqlNativePassword(user.Auth1.Password)
+		} else if len(user.Auth1.Plugin) > 0 {
+			authUser.Auth1 = plan.NewOtherAuthentication(user.Auth1.Password, user.Auth1.Plugin)
+		} else {
+			// We default to using the password, even if it's empty
+			authUser.Auth1 = plan.NewDefaultAuthentication(user.Auth1.Password)
+		}
+	}
+	if user.Auth2 != nil || user.Auth3 != nil || user.AuthInitial != nil {
```

Comment:
if know you didn't write this and I don't know if this scenario is possible, but if there are multiple auth options this fails even if Auth1 is OK

Reply:
I dug into this a little bit and it looks like this is by design. This line is the only reference to `Auth2`, `Auth3`, and `AuthInitial` in GMS, so this seems to just be ensuring that only `Auth1` is ever used. 

I went ahead and added a comment mentioning that, too.